ansible_collections/arista/avd/roles/eos_designs/python_modules/network_services/ethernet_interfaces.py

```diff
@@ -36,17 +37,33 @@ def ethernet_interfaces(self) -> dict | None:
                     # The l3_interfaces has already been filtered in _filtered_tenants
                     # to only contain entries with our hostname
                     for l3_interface in vrf["l3_interfaces"]:
+                        nodes_length = len(l3_interface["nodes"])
+                        if (
+                            len(l3_interface["interfaces"]) != nodes_length
+                            or len(l3_interface["ip_addresses"]) != nodes_length
+                            or ("descriptions" in l3_interface and "description" not in l3_interface and len(l3_interface["descriptions"]) != nodes_length)
+                        ):
+                            raise AristaAvdError(
+                                "Length of lists 'interfaces', 'nodes', 'ip_addresses' and 'descriptions' (if used) must match for l3_interfaces for"
+                                f" {vrf['name']} in {tenant['name']}"
+                            )
+
                         for node_index, node_name in enumerate(l3_interface["nodes"]):
                             if node_name != self._hostname:
                                 continue
 
                             interface_name = str(l3_interface["interfaces"][node_index])
```

Comment:
Would it not make more sense to do something like:
```suggestion
                        node_index = l3_interface["nodes].index(self._hostname)
                        interface_name = str(l3_interface["interfaces"][node_index])
```
(and then fix the rest of the indentation below).  You would obviously need to catch an IndexError exception.

Reply:
This is not even needed. Since we walk over the filtered_tenants stuff, everything is already filtered to only contain entries that are relevant to us.

Reply:
will remove this!

Reply:
hmm
```
TASK [arista.avd.eos_designs : Generate device configuration in structured format] ***
An exception occurred during task execution. To see the full traceback, use -vvv. The error was: Exception: {'interfaces': ['Ethernet10', 'Ethernet10'], 'ip_addresses': ['10.10.30.10/24', '10.10.40.20/24'], 'nodes': ['DC1-BL1A', 'DC1-BL1B'], 'mtu': 9000, 'enabled': True, 'descriptions': ['test-DC1-BL1A', 'test-DC1-BL1B']}

````

Looks like it is needed actually

so will apply @jonxstill  suggestion

Reply:
I was referring to the catch for index error, but actually your first loop is better, since we support having multiple interfaces on the same device. So you will have to loop over them.

Reply:
I am sorry but  I am not exactly getting what is required here. I am kind of thinking that it should not be changed from what I have actually. I have thought about it and the current code allow to properly handle the situation where a device appears multiple time in the list.

@jonxstill's solution with index would only return the first occurence every time and applying it made some changes to the output config.

@ClausHolbechArista when you say "catch for index error" what line are you referring to? I have been testing the current code I cannot see how to remove any line. If I raise in this for loop I get outputs like:

```
 {'interfaces': ['Ethernet8', 'Ethernet9', 'Ethernet8', 'Ethernet9'], 'ip_addresses': ['10.10.10.10/24', '10.10.20.20/24', '10.10.30.10/24', '10.10.40.20/24'], 'nodes': ['DC1-BL1A', 'DC1-BL1A', 'DC1-BL1B', 'DC1-BL1B'], 'mtu': 9000, 'enabled': True, 'description': 'test'}
```

So I still need to check for the hostname.

Am I missing something here?

Reply:
I was not clear. I was referring to the last line of Jon's initial comment `You would obviously need to catch an IndexError exception.`.
Keep the loop as is.

Reply:
ah yeah I was completely not getting that part! Apologies! - if everybody is ok I would probably suggest to resolve this open comment?